mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -38,6 +40,19 @@ impl RegisterTokenInfo<TendermintToken> for TendermintCoin {
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum TendermintPubkeyActivationParams {
+    /// Activation via public key
+    WithPubkey {
+        #[serde(deserialize_with = "deserialize_account_public_key")]
+        pubkey: TendermintPublicKey,
+        is_ledger_connection: bool,
+    },
+    /// Activation via WalletConnect
+    WalletConnect,
```

Comment:
Could you add some info about why we don't take public key similar to how it's done with `WithPubkey` variant?

Also, I'm curious why `WithPubkey` mode doesn’t work here. Doesn’t WalletConnect already provide the public key via an API? `WithPubkey` mode was intended to support **any** type of external wallet as it only requires the public key and doesn’t depend on the wallet’s internal logic.

Reply:
`WalletConnect` doesn’t directly provide the public key as part of the initial session information. Instead, it only returns the account address, methods and the chains the user can interact with. To obtain the public key, a separate request is required for the specific account

Reply:
I would have done it in a way where some library/crate (or even GUI side) handles this and then utilizes `with_pubkey` on core KDF side to have single entrypoint for any external wallet, so the complexity remains constant for N external wallet types.

Reply:
this is like dividing this implementation into two (manages connection from GUI while kdf manages request handling )  or leaving everything entirely to `GUI` which is not what we wanted from the beginning. 

So the thing is whoever wishes to use any wallet pubkey via WalletConnect needs to make an additional request which can't really be avoided.

Reply:
What I mean was, we could handle that connection part and the extra request part (for getting pubkey) in a separate crate and utilize the `with_pubkey` mode. I mean, we have an abstraction layer for external wallets, so why not using it?

Reply:
> how do you plan to do this? since, we either ways we still need to request for pubkey meanwhile with_pubkey params requires pubkey in activation params itself. There's a [kdf_walletconnect](https://github.com/KomodoPlatform/komodo-defi-framework/tree/59db165d9fd99e8cb98ac30e33dd7a691b40e89d/mm2src/kdf_walletconnect) crate that handles all WC logic By the way,.

User sends the activation request for WC -> KDF takes the request and runs the logic where it reads Pubkey from WC -> Uses that Pubkey to run KDF as an external wallet (just like any other wallet).

Is that not possible?

> Also, the current external wallet implementation for tendermint uses SSE, in the case of Keplr, we need to send request to GUI then to Keplr wallet if I'm correct, so it can't directly work with WalletConnect. The current activation logic is quite easy to understand and scale(when integrating KDF as Wallet in the future)

Yeah, for Keplr, GUI side triggers the Keplr events which when required. But it doesn't have to be the same way for WalletConnect.

Reply:
> User sends the activation request for WC -> KDF takes the request and runs the logic where it reads Pubkey from WC -> Uses that Pubkey to run KDF as an external wallet (just like any other wallet).

that's how the current flow works.. or am getting something wrong..
User request coin activation, logic checks if `TendermintPubkeyActivationParams::WalletConnect`
https://github.com/KomodoPlatform/komodo-defi-framework/blob/f5caa1918b437b2593e55013a270f4b83f210c21/mm2src/coins_activation/src/tendermint_with_assets_activation.rs#L306-L308

then KDF takes the request and runs the logic where it reads Pubkey from WC( maybe this function should have a different naming)
https://github.com/KomodoPlatform/komodo-defi-framework/blob/f5caa1918b437b2593e55013a270f4b83f210c21/mm2src/coins_activation/src/tendermint_with_assets_activation.rs#L232-L260

Uses that Pubkey to run KDF as an external wallet (just like any other wallet).
https://github.com/KomodoPlatform/komodo-defi-framework/blob/f5caa1918b437b2593e55013a270f4b83f210c21/mm2src/coins_activation/src/tendermint_with_assets_activation.rs#L259